layers/+lang/typescript/packages.el

```diff
@@ -2,21 +2,42 @@
 ;;
 ;; Copyright (c) 2012-2016 Sylvain Benner & Contributors
 ;;
-;; Author: Chris Bowdon <c.bowdon@bath.edu>
-;; URL: https://github.com/syl20bnr/spacemacs
+;; Author: JAremko <w3techplaygound@gmail.com> 
+;; URL: https://github.com/JAremko/spacemacs-pr
 ;;
 ;; This file is not part of GNU Emacs.
 ;;
 ;;; License: GPLv3
 
-(setq typescript-packages '(tss))
+(setq typescript-packages '(tide))
 
-(defun typescript/init-tss ()
-  "Initialize my package"
-  (use-package tss
+(defun typescript/init-tide ()
+  (use-package tide
     :defer t
-    :mode ("\\.ts\\'" . typescript-mode)
-    :init (spacemacs/set-leader-keys-for-major-mode 'typescript-mode
-            "gg" 'tss-jump-to-definition
-            "hh" 'tss-popup-help)
-    :config (tss-config-default)))
+    :init (progn
+            (evilified-state-evilify tide-references-mode tide-references-mode-map
+              (kbd "C-j") 'tide-find-previous-reference
+              (kbd "C-k") 'tide-find-next-reference
+              (kbd "C-l") 'tide-goto-reference)
+            (add-hook 'typescript-mode-hook
+              (lambda ()
+                (tide-setup)
+                (flycheck-mode t)
+                (setq flycheck-check-syntax-automatically '(save mode-enabled))
+                (eldoc-mode t)
+                (when (configuration-layer/package-usedp 'company) 
+                  (company-mode-on)))))
+    :config (progn
+              (spacemacs/declare-prefix-for-mode 'typescript-mode "mg" "goto")
+              (spacemacs/declare-prefix-for-mode 'typescript-mode "mh" "help")
+              (spacemacs/declare-prefix-for-mode 'typescript-mode "mn" "name")
+              (defun typescript-jump-to-type-definition ()
```

Comment:
The [docs](https://github.com/syl20bnr/spacemacs/blob/master/doc/LAYERS.org#configel-funcsel-and-keybindingsel) say:
> `funcs.el` definition of *general* utility functions  

and

> The word general here means independent of any package. Since the end user can exclude an arbitrary set of packages, you cannot be sure that, just because your layer includes a package, that package will necessarily be loaded. For this reason, code in these files must be generally safe, regardless of which packages are installed.

Since it's just a wrapper function for `tide-jump-to-definition` I'll keep it in the `packages.el`